### PR TITLE
appveyor: Generate PDBs for mingw build

### PR DESCRIPTION
**Pull request by MerryMage**
This PR generates PDBs for the mingw build. It does this by first generating debug information in the executable, then stripping it and putting it in a PDB file with cv2pdb. URL to that executable will be updated once citra-emu/ext-windows-bin#13 is merged.

In addition to generating PDBs, we also embed Source Server information. This would enable MSVC to retrieve the exact version of the source files that were used to build an application from github directly when you load up a compiled application or a minidump in the debugger.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/3172)

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -71,6 +71,49 @@ after_build:
           $RELEASE_DIST = "head"
         }
 
+        # Generate pdb files for mingw
+        if ($env:BUILD_TYPE -eq 'mingw') {
+          Invoke-WebRequest -Uri https://raw.githubusercontent.com/MerryMage/ext-windows-bin/master/cv2pdb/cv2pdb.exe -OutFile cv2pdb.exe
```
Comment (jroweboy) [inline appveyor.yml]: This should be PRed and merged on ext-windows-bin

File: appveyor.yml [hunk 2]
```diff
@@ -93,21 +136,22 @@ after_build:
           7z a $MSVC_SEVENZIP $RELEASE_DIST
         } else {
           $MINGW_BUILD_ZIP = "citra-windows-mingw-$GITDATE-$GITREV.zip" -replace " ", ""
+          $MINGW_BUILD_PDB = "citra-windows-mingw-$GITDATE-$GITREV-debugsymbols.zip" -replace " ", ""
           $MINGW_SEVENZIP = "citra-windows-mingw-$GITDATE-$GITREV.7z" -replace " ", ""
           # not going to bother adding separate debug symbols for mingw, so just upload a README for it
```
Comment (jroweboy) [inline appveyor.yml]: these comments should be removed as well

File: appveyor.yml (hunk 1 above)
Comment (MerryMage) [reply appveyor.yml]: Waiting on citra-emu/ext-windows-bin#13.

**Comment by MerryMage** [comment]
One thing I've noted when I've tried to debug an executable is for the best experience you'll want to have a local mingw install since the srcsrv doesn't include mingw headers.

File: appveyor.yml [hunk 3]
```diff
@@ -71,6 +71,49 @@ after_build:
...
+        foreach ($repo in @{
+          "citra-emu/citra" = ""
+          "citra-emu/ext-boost" = "externals/boost"
+          "citra-emu/ext-soundtouch" = "externals/soundtouch"
+          "fmtlib/fmt" = "externals/fmt"
+          "herumi/xbyak" = "externals/xbyak"
+          "lsalzman/enet" = "externals/enet"
+          "MerryMage/dynarmic" = "externals/dynarmic"
+          "neobrain/nihstro" = "externals/nihstro"
+          "svn2github/inih" = "externals/inih/inih"
```
Comment (MerryMage) [inline appveyor.yml]: `benhoyt/inih`

**Comment by wwylele** [comment]
What's the status of this? Is this ready to merge?

**Comment by MerryMage** [comment]
Not yet, I want to fix that line above and double check it still works since we've made some appveyor build changes since this PR.

**Comment by B3n30** [comment]
@bunnei you reviewed that PR, but there were changes since your last review. Do you want to give it a second look?

@jroweboy I think it would be great if you could give this PR an approval (if everything is fine), so we can merge it

**Comment by MerryMage** [comment]
Sorry for the delay. Fixed the above. 

Also made pdb extraction happen *after* getting dependencies.

File: .appveyor/ProcessPdb.ps1 [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  Invoke-WebRequest -Uri https://raw.githubusercontent.com/citra-emu/ext-windows-bin/master/cv2pdb/cv2pdb.exe -OutFile cv2pdb.exe
+  foreach ($exe in Get-ChildItem "$RELEASE_DIST" -Recurse -Filter "citra*.exe") {
+    .\cv2pdb $exe.FullName
+  }
+}
+
+# Specify source locations in pdb via srcsrv.ini
+$srcsrv = "SRCSRV: ini ------------------------------------------------`r`n"
+$srcsrv += "VERSION=2`r`n"
+$srcsrv += "VERCTRL=http`r`n"
```
Comment (jroweboy) [inline .appveyor/ProcessPdb.ps1]: is this supposed to be http? (i'm guessing its something that ms defines as both http and https?)

File: .appveyor/ProcessPdb.ps1 [hunk 5]
```diff
@@ -0,0 +1,42 @@
+# Generate pdb files for mingw
+if ($env:BUILD_TYPE -eq 'mingw') {
+  Invoke-WebRequest -Uri https://raw.githubusercontent.com/citra-emu/ext-windows-bin/master/cv2pdb/cv2pdb.exe -OutFile cv2pdb.exe
+  foreach ($exe in Get-ChildItem "$RELEASE_DIST" -Recurse -Filter "citra*.exe") {
+    .\cv2pdb $exe.FullName
```
Comment (jroweboy) [inline .appveyor/ProcessPdb.ps1]: should we strip symbols out of the binary afterwards to save space? (how much extra MB is added for compiling with `-g`?)

Comment (MerryMage) [reply .appveyor/ProcessPdb.ps1]: cv2pdb strips symbols itself.

File: .appveyor/ProcessPdb.ps1 (hunk 4 above)
Comment (MerryMage) [reply .appveyor/ProcessPdb.ps1]: It's supposed to be `http`. It describes the type of version control system to retrieve the source code from. (i.e.: Subversion vs Team vs http).

File: .appveyor/ProcessPdb.ps1 [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  "citra-emu/citra" = ""
+  "citra-emu/ext-boost" = "externals/boost"
+  "citra-emu/ext-soundtouch" = "externals/soundtouch"
+  "fmtlib/fmt" = "externals/fmt"
+  "herumi/xbyak" = "externals/xbyak"
+  "lsalzman/enet" = "externals/enet"
+  "MerryMage/dynarmic" = "externals/dynarmic"
+  "neobrain/nihstro" = "externals/nihstro"
+  "weidai11/cryptopp" = "externals/cryptopp/cryptopp"
+  "whoshuu/cpr" = "externals/cpr"
```
Comment (jroweboy) [inline .appveyor/ProcessPdb.ps1]: Oops! Forgot to mention we removed cpr and replaced it with a few other libs recently. Need to change this to point to the new libs. (Or maybe just strip the list down to a few core external depenedencies?)

File: appveyor.yml [hunk 7]
```diff
@@ -130,6 +130,10 @@ after_build:
           # copy the qt windows plugin dll to platforms
           Copy-Item -path "C:/msys64/mingw64/share/qt5/plugins/platforms/qwindows.dll" -force -destination "$RELEASE_DIST/platforms"
 
+          # process PDBs
+          . "./.appveyor/ProcessPdb.ps1"
+
+          7z a -tzip $MINGW_BUILD_PDB $RELEASE_DIST\*.pdb
```
Comment (jroweboy) [inline appveyor.yml]: We should remove the PDBs after this line, so we don't have to increase the zip size by ~8MB

**Pull request merged**